tree/ntuple/v7/test/ntuple_types.cxx

```diff
@@ -1166,6 +1166,106 @@ TYPED_TEST(UniquePtr, Basics)
    EXPECT_EQ(nullptr, pArray->get());
 }
 
+TEST(RNTuple, Optional)
```

Comment:
Are we missing test for:
```
std::optional<std::string>
std::optional<std::array<...>>
std::optional<std::map<...>>
etc.
```
and
```
std::optional<std::variant<...>>
```
?

Reply:
With #15582 we have a test for `std::optional<std::variant<...>>`. I'll add test for the others.

Reply:
Did we have to make adjustment to the `std::optional` support?

Reply:
For `std::optional<std::variant<...>>`>? No.

Reply:
I realize, `std::optional<std::array<...>>` is also already tested.

Reply:
https://github.com/root-project/root/pull/15715